server/server.go

```diff
@@ -355,7 +355,35 @@ func (s *Server) Listeners() ([]Loop, error) {
 	return loops, nil
 }
 
+// Addrs returns a list of addresses that the server is listening on.
+// if the server hasn't been started it will not return an address.
+func (s *Server) Addrs() []string {
+	var addrs []string
+	for _, l := range s.httpListeners {
+		a := l.Addr()
+		if a != "" {
+			addrs = append(addrs, a)
+		}
+	}
+	return addrs
+}
+
+type tcpKeepAliveListener struct {
+	*net.TCPListener
+}
+
+func (ln tcpKeepAliveListener) Accept() (net.Conn, error) {
+	tc, err := ln.AcceptTCP()
+	if err != nil {
+		return nil, err
+	}
+	tc.SetKeepAlive(true)
+	tc.SetKeepAlivePeriod(3 * time.Minute)
```

Comment:
NIT: use a const for this instead of 3.

Reply:
I left it as-is to match https://golang.org/src/net/http/server.go?s=86961:87002#L3263 I figure the least amount of deviation the better.